### PR TITLE
Add detection rule for OpenAI tokens

### PR DESCRIPTION
**Pull request by mkthebea**
### Description:
This PR adds detection rule for OpenAI tokens.

### Checklist:

* [x] Does your PR pass tests?
* [x] Have you written new tests for your changes?
* [x] Have you lint your code locally prior to submission?

### CODE REVIEW

File: cmd/generate/config/rules/openai.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  
+  "github.com/zricethezav/gitleaks/v8/config"
+)
+
+func OpenAI() *config.Rule {
+    // Define Rule
+    r := config.Rule{
+        Description: "Open AI token",
+        RuleID:      "open-ai-token",
+        Regex: regexp.MustCompile("(sk-)[a-zA-Z0-9]{48}"),
```
[the remark this answers is not in the document]
Comment (debasisdwivedy) [reply cmd/generate/config/rules/openai.go:14]: Any updates on this

[the remark this answers is not in the document]
Comment (mkthebea) [reply cmd/generate/config/rules/openai.go]: Thank you. As you said, I added a more specific pattern for openAI token detection.

File: cmd/generate/config/rules/openai.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func OpenAI() *config.Rule {
+    // Define Rule
+    r := config.Rule{
+        Description: "Open AI token",
+        RuleID:      "open-ai-token",
+        Regex: regexp.MustCompile("(sk-)[a-zA-Z0-9]{20}(T3BlbkFJ)[a-zA-Z0-9]{20}"),
```
Comment (zricethezav) [inline cmd/generate/config/rules/openai.go:14]: `sk` isn't restrictive enough. Please use `generateSemiGenericRegex`

> `T3BlbkFJ`

What is this?

Comment (mkthebea) [reply cmd/generate/config/rules/openai.go:14]: I've repeatedly issued tokens on openAI and found that all tokens have a common "T3BlbkFJ" and that there exists a random string with a length of 20 before and after that.

Comment (Zoheb-Malik) [reply cmd/generate/config/rules/openai.go]: Hi @mkthebea, is it the same case if an account is using the GPT-4 model? Or is "T3BlbkFJ" common amongst all keys, iregardless of the model? Thanks!

Comment (rgmz) [reply cmd/generate/config/rules/openai.go]: They are "OpenAI" tokens and not specific to any specific service, so the model shouldn't matter.

**Pull request closed without merging**